db/python/tables/bq/billing_daily_extended.py

```diff
@@ -49,3 +53,317 @@ async def get_extended_values(self, field: str):
 
         # return empty list if no record found
         return []
+
+    async def get_batch_cost_summary(
+        self,
+        start_time: datetime,
+        end_time: datetime,
+        batch_ids: list[str] | None,
+        ar_guid: str | None,
+    ) -> list[BillingBatchCostRecord]:
+        """
+        Get summary of AR run
+        """
+        query_parameters = [
+            bigquery.ScalarQueryParameter('start_time', 'TIMESTAMP', start_time),
+            bigquery.ScalarQueryParameter('end_time', 'TIMESTAMP', end_time),
+        ]
+
+        if ar_guid:
+            condition = 'ar_guid = @ar_guid'
+            query_parameters.append(
+                bigquery.ScalarQueryParameter('ar_guid', 'STRING', ar_guid)
+            )
+        elif batch_ids:
+            condition = 'batch_id in UNNEST(@batch_ids)'
+            query_parameters.append(
+                bigquery.ArrayQueryParameter('batch_ids', 'STRING', batch_ids)
+            )
+        else:
+            raise ValueError('Either ar_guid or batch_ids must be provided')
+
+        # The query will locate all the records for the given time range
+        # and batch_ids or ar_guid
+        # getting all records in one go to limit number of queries,
+        # cutting the total cost of BQ
+        # After getting all the needed data,
+        # aggregate:
+        # the cost per topic,
+        # get counts of batches,
+        # cost per hail, cromwell and dataproc
+        # start and end date of the usage
+        # total cost,
+        # including cost by sku
```

Comment:
This text feels a little strangely formatted, maybe needs more indentation if they're list items.